File: Microservices/BillingService/BillingService.Domain/Models/Claims/ClaimProcessRequestModel.cs
using System.Diagnostics.CodeAnalysis;

namespace BillingService.Domain.Models.Claims
{
    [ExcludeFromCodeCoverage]
    public class ClaimProcessRequestModel
    {
        public string BatchId { get; set; }
        public ClaimsSubmitModel RequestModel { get; set; }
        public int TotalClaims { get; set; }
        public string? ClaimStatus { get; set; }
    }
    [ExcludeFromCodeCoverage]
    public class ClaimApproveRequestModel
    {
        public string BatchId { get; set; }
        public IdsWithUserInfo RequestModel { get; set; }
        public int TotalClaims { get; set; }
    }
}

File: billing_ui_app/src/app/core/models/billing/claim-process-request-model.ts
import { ClaimsSubmitModel } from "./claims-submit-model";
import { IdsWithUserInfo } from "./get-claim-by-identifier";

export class ClaimProcessRequestModel {
    batchId : string;
    claimsSubmitModel: ClaimsSubmitModel[] = [];
    totalClaims : number;
    claimStatus: string;
}

export class ClaimApproveRequestModel extends IdsWithUserInfo{
    batchId : string;
    totalClaims : number;

}
